pkg/multitenant/multitenant.go

```diff
@@ -31,7 +38,14 @@ func (s singleTenant) GetIDs(context.Context) ([]string, error) {
 }
 
 func WithTenant(ctx context.Context, tenantId string) context.Context {
-	return context.WithValue(ctx, tenantCtx{}, tenantId)
+	if span := trace.SpanFromContext(ctx); span.IsRecording() {
+		span.SetAttributes(attribute.String(tracingAttributeName, tenantId))
+	}
+
+	return kuma_log.WithFields(
```

Comment:
Can we name this function `CtxWithFields` or `CtxWithLogFields`? When I saw `kuma_log.WithFields` my guess was that we return a logger with fields (since the package is about logging), but that's not true.